### DISCUSSION TITLE
How do I add extensions locally?

### DISCUSSION THREAD

**Discussion by KrisCoulson** [supabase/supabase#2915] [Questions]
After reading the pg_cron article https://supabase.io/blog/2021/03/05/postgres-as-a-cron-server I was wondering how to run pg_cron locally. In the default supabase containers pg_cron isn't installed by default and it seems that postgres version is 12.3.

I found the supabase/postgres image which says if your use it you will get a bunch of added default extensions and it uses postgres 13 so I added that image to my .supabase/docker/docker-compose.yml  ran `supabase start` thinking that the container would now be running that image but I don't think it is. Version is still at 12.3 and the extensions aren't added

Not just this extension but how do we add extensions to locally hosted supabase to test functionality before putting it on our prod db?

**Comment by dthyresson** [comment]
Not an answer, but I had the same question as @KrisCoulson and when I tried to

```
select name, comment, default_version, installed_version
from pg_available_extensions
where name = 'pg_cron';
```

I was informed that it wasn't available.

If I do try to create the extension:

```
Query 1 ERROR: ERROR:  could not open extension control file "/usr/share/postgresql/12/extension/pg_cron.control": 
No such file or directory
```
